server/src/internalClusterTest/java/org/opensearch/indices/IndicesRequestCacheDiskTierIT.java

```diff
@@ -111,7 +113,257 @@ public void testDiskTierStats() throws Exception {
+            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0);
+
+        assertAcked(
+            client.admin().indices().prepareCreate("index").setMapping("k", "type=keyword").setSettings(indicesSettingBuilder).get()
+        );
+
+        indexRandom(true, client.prepareIndex("index").setSource("k", "hello"));
+        ensureSearchable("index");
+        SearchResponse resp;
+
+        resp = client.prepareSearch("index").setRequestCache(true).setQuery(QueryBuilders.termQuery("k", "hello" + 0)).get();
+        int requestSize = (int) getCacheSizeBytes(client, "index", TierType.ON_HEAP);
+        assertTrue(heapSizeBytes > requestSize);
+        // If this fails, increase heapSizeBytes! We can't adjust it after getting the size of one query
+        // as the cache size setting is not dynamic
+        int numOnDisk = 2;
+        int numRequests = heapSizeBytes / requestSize + numOnDisk;
+        RequestCacheStats requestCacheStats = client.admin()
+            .indices()
+            .prepareStats("index")
+            .setRequestCache(true)
+            .get()
+            .getTotal()
+            .getRequestCache();
+        for (int i = 1; i < numRequests; i++) {
+            requestCacheStats = client.admin()
+                .indices()
+                .prepareStats("index")
+                .setRequestCache(true)
+                .get()
+                .getTotal()
+                .getRequestCache();
+            resp = client.prepareSearch("index").setRequestCache(true).setQuery(QueryBuilders.termQuery("k", "hello" + i)).get();
+            assertSearchResponse(resp);
+            IndicesRequestCacheIT.assertCacheState(client, "index", 0, i + 1, TierType.ON_HEAP, false);
+            IndicesRequestCacheIT.assertCacheState(client, "index", 0, i + 1, TierType.DISK, false);
+        }
+
+        requestCacheStats = client.admin()
+            .indices()
+            .prepareStats("index")
+            .setRequestCache(true)
+            .get()
+            .getTotal()
+            .getRequestCache();
+
+        long entries = requestCacheStats.getEntries(TierType.DISK);
+        // make sure we have 2 entries in disk.
+        assertEquals(2, entries);
+
+        // call clear cache api
+        client.admin().indices().prepareClearCache().setIndices("index").setRequestCache(true).get();
+        // fetch the stats again
+        requestCacheStats = client.admin()
+            .indices()
+            .prepareStats("index")
+            .setRequestCache(true)
+            .get()
+            .getTotal()
+            .getRequestCache();
+        entries = requestCacheStats.getEntries(TierType.DISK);
+        // make sure we have 0 entries in disk.
+        assertEquals(0, entries);
+    }
+
+    // When entire disk tier is stale, test whether cache cleaner cleans up everything from disk
+    public void testDiskTierInvalidationByCacheCleanerEntireDiskTier() throws Exception {
+        int thresholdInMillis = 4_000;
```

Comment:
Set lower so the test doesn't take so long to run? Or does it flake if you do that